spark-commons/src/test/scala/za/co/absa/spark/commons/OncePerSparkSessionTest.scala

```diff
@@ -48,4 +48,41 @@ class OncePerSparkSessionTest extends AnyFunSuite with MockitoSugar with SparkTe
     assert(libraryBInitCounter == 2)
   }
 
+  test("should return true if the library is registered successfully and false if not and if spark is not hasn't started"){
+    class UDFLibrary()(implicit sparkToRegister: SparkSession) extends OncePerSparkSession(){
+
+      var libraryAInitCounter = 0
+      val results_1 = this.register(sparkToRegister)
+      val results_2 = this.register(sparkToRegister)
+      override protected def registerBody(spark: SparkSession): Unit = {
+        this.libraryAInitCounter += 1
+      }
+
+    }
+
+    val UdfLibrary = new UDFLibrary()
+    assert(UdfLibrary.results_1 == true)
+    assert(UdfLibrary.results_2 == false)
+    assert(UdfLibrary.libraryAInitCounter == 1)
+  }
+
+  test("should return false if the library is registered successfully and false if provided that spark session has started") {
```

Comment:
Not sure of this tests purpose, at least the description is not very explanatory.
Could be testing auto-registering, so:
* have a spark in constructor
* use the parent's constructor with spark, therefore auto-registering
* then call `register` in the body of the test asserting it's returns negative
* then call `register` again with the mock spark and assert it being true